src/main/java/ch/uzh/ifi/hase/soprafs21/objects/GameEngine.java

```diff
@@ -219,18 +219,27 @@ private UUID inGameSession(User user){
     }
 
 
-    public void clearRequestsByHost(UUID hostID){gameSessionRequestList.clearByHostID(hostID);}
+    public void clearRequestsByHost(Long hostID){gameSessionRequestList.clearByHostID(hostID);}
 
-    public void clearRequestByUser(UUID userID, UUID gameSessionID){gameSessionRequestList.clearByUserAndGameSessionID(userID,gameSessionID);}
+    public void clearRequestByUser(Long userID, UUID gameSessionID){gameSessionRequestList.clearByUserAndGameSessionID(gameSessionID, userID);}
 
 
     public void addRequest(GameSessionRequest gameSessionRequest){gameSessionRequestList.addRequest(gameSessionRequest);}
 
     public void addUserToSession(User user,UUID gameSessionID){
         try {
-            for (GameSession gameSession : gameSessionList) {
-                if (gameSession.getID().equals(gameSessionID)) {
-                    gameSession.addUser(user);
+            if(gameEngine.findGameSessionByID(gameSessionID).getUserList().size()==4){
+                gameEngine.createGameFromGameSession(findGameSessionByID(gameSessionID));
+
+            }else {
+                for (GameSession gameSession : gameSessionList) {
+                    if (gameSession.getID().equals(gameSessionID)) {
+                        gameSession.addUser(user);
+                    }
+                }
+                if(gameEngine.findGameSessionByID(gameSessionID).getUserList().size()==4) {
```

Comment:
replace 4 with PLAYER_AMOUNT (already exists as private variable on the game engine), that we can easily update the player amount for a game